CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLWorker.cc

```diff
@@ -0,0 +1,855 @@
+#include "TrackingTools/TransientTrack/interface/TransientTrack.h"
+#include "CalibTracker/Records/interface/SiPixelTemplateDBObjectESProducerRcd.h"
+#include "CondFormats/SiPixelObjects/interface/SiPixelTemplateDBObject.h"
+#include "CondFormats/SiPixelTransient/interface/SiPixelTemplateDefs.h"
+#include "CondFormats/SiPixelTransient/interface/SiPixelTemplate.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelTopologyMap.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "../interface/SiPixelLorentzAngleCalibrationStruct.h"
+#include <TTree.h>
+#include <TFile.h>
+#include <fstream>
+//
+// class declaration
+//
+
+static const int maxpix = 1000;
+struct Pixinfo {
+  int npix;
+  float row[maxpix];
+  float col[maxpix];
+  float adc[maxpix];
+  float x[maxpix];
+  float y[maxpix];
+};
+
+struct Hit {
+  float x;
+  float y;
+  double alpha;
+  double beta;
+  double gamma;
+};
+struct Clust {
+  float x;
+  float y;
+  float charge;
+  int size_x;
+  int size_y;
+  int maxPixelCol;
+  int maxPixelRow;
+  int minPixelCol;
+  int minPixelRow;
+};
+struct Rechit {
+  float x;
+  float y;
+};
+
+class SiPixelLorentzAnglePCLWorker : public DQMGlobalEDAnalyzer<SiPixelLorentzAngleCalibrationHistograms> {
+public:
+  explicit SiPixelLorentzAnglePCLWorker(const edm::ParameterSet&);
+  ~SiPixelLorentzAnglePCLWorker() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void bookHistograms(DQMStore::IBooker&,
+                      edm::Run const&,
+                      edm::EventSetup const&,
+                      SiPixelLorentzAngleCalibrationHistograms&) const override;
+
+  void dqmAnalyze(edm::Event const&,
+                  edm::EventSetup const&,
+                  SiPixelLorentzAngleCalibrationHistograms const&) const override;
+
+  void dqmBeginRun(edm::Run const&, edm::EventSetup const&, SiPixelLorentzAngleCalibrationHistograms&) const override;
+
+  void dqmEndRun(edm::Run const&,
+                 edm::EventSetup const&,
+                 SiPixelLorentzAngleCalibrationHistograms const&) const override;
+
+  const Pixinfo fillPix(const SiPixelCluster& LocPix, const PixelTopology* topol) const;
+  const std::pair<LocalPoint, LocalPoint> surface_deformation(const PixelTopology* topol,
+                                                              TrajectoryStateOnSurface& tsos,
+                                                              const SiPixelRecHit* recHitPix) const;
+  // ------------ member data ------------
+  std::string folder_;
+  bool notInPCL_;
+  std::string filename_;
+  std::vector<std::string> newmodulelist_;
+
+  // histogram etc
+  int hist_x_;
+  int hist_y_;
+  double min_x_;
+  double max_x_;
+  double min_y_;
+  double max_y_;
+  double width_;
+  double min_depth_;
+  double max_depth_;
+  double min_drift_;
+  double max_drift_;
+  int bufsize;
```

Comment:
```suggestion
  int bufsize_;
```

Reply:
Hmm I guess one can argue that this should stay `bufsize`, let's see what others think